File: ios/Classes/YSFPPProxy.m
//
//  YSFPPProxy.m
//  MacOSProxyResolver
//
//  Created by Sergey Yamshchikov on 07.02.21.
//  Copyright © 2021 Sergey Yamshchikov. All rights reserved.
//

#import "YSFPPProxy.h"

@implementation YSFPPProxy

@synthesize host = _host;
@synthesize port = _port;
@synthesize user = _user;
@synthesize password = _password;

- (instancetype)initWithHost:(nullable NSString *)host port:(nullable NSString *) port user: (nullable NSString *)user password: (nullable NSString *) password type:(YSFPPProxyType) type
{
    self = [super init];
    if (self) {
        _host = host;
        _port = port;
        _user = user;
        _password = password;
        _type = type;
    }
    return self;
}

- (NSString*)host {
    return [self toStringForString:_host];
}

- (NSString*)port {
    return [self toStringForString:_port];
}

- (NSString*)user {
    return [self toStringForString:_user];
}

- (NSString*)password {
    return [self toStringForString:_password];
}

- (NSString*)typeString {
    switch (self.type) {
        case ySFPPHTTPS:
            return @"https";
            break;
        case ySFPPHTTP:
            return @"http";
            break;
        case ySFPPNONE:
            return @"none";
            break;
    }
}

- (NSString*) toStringForString:(nullable NSString *)value {
    if (value) {
        return value;
    } else {
        return @"";
    }
}

- (NSString*)description {
    return [NSString stringWithFormat:@"{\"host\": \"%@\", \"port\": \"%@\", \"user\": \"%@\", \"password\": \"%@\", \"type\": \"%@\"}", self.host, self.port, self.user, self.password, [self typeString]];
}

@end

File: macos/Classes/YSFPPProxy.swift
import Foundation

class YSFPPProxy {
    let host: String?
    let port: String?
    let user: String?
    let password: String?
    let type: YSFPPProxyType
    
    init(host: String?, port: String?, user: String?, password: String?, type: YSFPPProxyType) {
        self.host = host
        self.port = port
        self.user = user
        self.password = password
        self.type = type
    }
    
    private func toString(_ value: String?) -> String {
        return value ?? ""
    }
    
    private var typeString: String {
        switch type {
        case .https: return "https"
        case .http: return "http"
        case .none: return "none"
        }
    }
    
    var description: String {
        return "{\"host\": \"\(toString(host))\", \"port\": \"\(toString(port))\", \"user\": \"\(toString(user))\", \"password\": \"\(toString(password))\", \"type\": \"\(typeString)\"}"
    }
}
